src/OmniSharp.Host/FileWatching/ManualFileSystemWatcher.cs

```diff
@@ -18,6 +20,16 @@ public void Notify(string filePath, FileChangeType changeType = FileChangeType.U
         {
             lock (_gate)
             {
+                if(changeType == FileChangeType.DirectoryDelete)
+                {
+                    foreach(var matchingCallback in _callbacksMap.AsEnumerable().Where(x => x.Key.StartsWith(filePath + Path.DirectorySeparatorChar, StringComparison.OrdinalIgnoreCase)))
```

Comment:
isn't this going to end up calling OnDocumentRemoved twice on a file? once from here as a file and then again through the directory?

Reply:
True, i will remove that part as it's unnecessary. 

Original idea were to hide directory removal specifics from subscribers of file watcher, however since it uses other than full paths (extensions etc) too it were not possible and this was left behind from that version.

Removed at https://github.com/OmniSharp/omnisharp-roslyn/pull/1821/commits/e90ea72d72e6b107785d2b11459dda1b75de9936